addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -68,8 +69,14 @@ def _l10n_eg_eta_connect_to_server(self, request_data, request_url, method, is_a
                 'error': str(ex),
                 'blocking_level': 'warning'
             }
-        if not request_response.ok:
+        try:
```

Comment:
This will probably work, but it makes the code less readable than it already is

maybe replace 
`if isinstance(response_data, dict) and response_data.get('error'):` below
with a try/except
as it already has some flawed logic like a default `_('Unknown error')` after it has already checked the dict for the key

(the author was trying to extract the error from the json response, which won't be available if the response is error code 400 or 500 or something similar)